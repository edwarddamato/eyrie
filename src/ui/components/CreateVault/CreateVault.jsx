import React from 'react';
import PropTypes from 'prop-types';
import { isNil } from 'ramda';
import { vault } from '../../../engine/vault/';
import { Form, Field, TextBox, Label, SubmitButton } from '../Form';
import './CreateVault.scss';
const remote = require('electron').remote;
const dialog = remote.require('electron').dialog;

class CreateVault extends React.Component {
  constructor (props) {
    super(props);

    this.state = {
      vaultCreated: ''
    };

    this.handleVaultFolderChoice = this.handleVaultFolderChoice.bind(this);
    this.handleCreateVault = this.handleCreateVault.bind(this);
  }

  handleVaultFolderChoice () {
    const loadingHandler = this.props.loadingHandler;
    loadingHandler(true);

    dialog.showOpenDialog({
      properties: ['openDirectory', 'createDirectory'],
      message: 'Choose where you want to create your vault'
    }, filepaths => {
      if (isNil(filepaths)) { loadingHandler(false); return; }
      this.setState({
        vaultCreated: filepaths[0]
      }, () => {
        loadingHandler(false);
      });
    });
  }

  handleCreateVault () {
    const loadingHandler = this.props.loadingHandler;
    const vaultLocation = this.state.vaultCreated;
    loadingHandler(true);

    if (isNil(vaultLocation)) { loadingHandler(false); return; }
    vault.create('my_vault', vaultLocation, 'secret123').then(() => {
      loadingHandler(false);
    });
  }

  render () {
    return (
      <section>
        <h2>Create Vault</h2>
        <Form onSubmit={values => { console.log(values); return false; }}>
          {({ onTextBoxChange, form }) => (
            <React.Fragment>
              <Field>
                <Label htmlFor="txtVaultName">Vault Name</Label>
                <TextBox id="txtVaultName" placeholder="Vault Name" onChange={onTextBoxChange} form={form} maxLength={25} />
              </Field>
              <Field>
                <Label htmlFor="txtVaultSecret">Vault Secret</Label>
                <TextBox id="txtVaultSecret" masked placeholder="Must be at least 8 characters long" onChange={onTextBoxChange} form={form} />
              </Field>
              <Field>
                <Label htmlFor="txtConfirmVaultSecret">Confirm Vault Secret</Label>
                <TextBox id="txtConfirmVaultSecret" masked placeholder="" onChange={onTextBoxChange} form={form} />
              </Field>
              <Field>
                <Label htmlFor="txtVaultLocation">Vault Location</Label>
                <TextBox
                  id="txtVaultLocation"
                  readonly
                  placeholder="Choose Folder..."
                  onClick={this.handleVaultFolderChoice}
                  value={this.state.vaultCreated}
                  onChange={onTextBoxChange}
                  form={form} />
              </Field>
              <SubmitButton text="Create" />
            </React.Fragment>
          )}
        </Form>
      </section>
    );
  }
}
CreateVault.propTypes = {
  loadingHandler: PropTypes.func.isRequired
};

export default CreateVault;
