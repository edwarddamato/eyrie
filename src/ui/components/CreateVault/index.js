import CreateVault from './CreateVault.jsx';
export default CreateVault;
