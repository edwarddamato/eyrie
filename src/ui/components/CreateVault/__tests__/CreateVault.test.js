import React from 'react';
import renderer from 'react-test-renderer';
import CreateVault from '../';

jest.mock('electron', () => {
  return {
    remote: {
      require: () => {
        return {
          dialog: {

          }
        };
      }
    }
  };
});

describe('CreateVault', () => {
  it('renders correctly', () => {
    const createVault = renderer.create(
      <CreateVault loadingHandler={() => { }} />
    ).toJSON();
    expect(createVault).toMatchSnapshot();
  });
});
