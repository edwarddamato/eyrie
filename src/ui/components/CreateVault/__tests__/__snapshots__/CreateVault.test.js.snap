// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateVault renders correctly 1`] = `
<section>
  <h2>
    Create Vault
  </h2>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form_field"
    >
      <label
        className="form_label"
        htmlFor="txtVaultName"
      >
        Vault Name
      </label>
      <input
        className="form_textbox"
        id="txtVaultName"
        maxLength={25}
        onChange={[Function]}
        placeholder="Vault Name"
        readOnly={undefined}
        type="text"
        value={undefined}
      />
    </div>
    <div
      className="form_field"
    >
      <label
        className="form_label"
        htmlFor="txtVaultSecret"
      >
        Vault Secret
      </label>
      <input
        className="form_textbox"
        id="txtVaultSecret"
        onChange={[Function]}
        placeholder="Must be at least 8 characters long"
        readOnly={undefined}
        type="password"
        value={undefined}
      />
    </div>
    <div
      className="form_field"
    >
      <label
        className="form_label"
        htmlFor="txtConfirmVaultSecret"
      >
        Confirm Vault Secret
      </label>
      <input
        className="form_textbox"
        id="txtConfirmVaultSecret"
        onChange={[Function]}
        placeholder=""
        readOnly={undefined}
        type="password"
        value={undefined}
      />
    </div>
    <div
      className="form_field"
    >
      <label
        className="form_label"
        htmlFor="txtVaultLocation"
      >
        Vault Location
      </label>
      <input
        className="form_textbox"
        id="txtVaultLocation"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Choose Folder..."
        readOnly={true}
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Create
    </button>
  </form>
</section>
`;
